rules/internal/bazel_deps_internal.bzl

```diff
@@ -0,0 +1,34 @@
+def scala_import_implementation(ctx):
+    if (ctx.attr.exports != []): fail("exports not supported yet")
+
+    compile_time_jars = depset(transitive = [
+        depset(direct = [
+            file for file in jar.files
+            if not file.basename.endswith("-sources.jar")
+        ])
+        for jar in ctx.attr.jars
+    ])
+
+    transitive_compile_time_jars = depset(transitive = [
+        entry[JavaInfo].transitive_compile_time_jars
+        for entry in ctx.attr.deps
+        if JavaInfo in entry
+    ])
+
+    runtime_jars = depset(transitive = [
+        entry[JavaInfo].transitive_runtime_jars
+        for entry in ctx.attr.runtime_deps
+        if JavaInfo in entry
+    ])
+
+    # TODO:
+    # consider exposing a ScalaInfo provider here too
+    # we'd need to infer the scala version from the jar file names
```

Comment:
What does rules_scala do?

Reply:
What do you mean?

Reply:
Ah, I just don't know much about this. I know rules_scala has a provider of some sort. Was just curious what it it did.

Reply:
Aah, got it. `rules_scala` just exposes a `JavaInfo` provider via the `java_common.create_provider` function. The JavaInfo provider will probably be required for IntelliJ support. I'd like to add a ScalaInfo provider too for cross compilation support (initially it will just allow us to emit nice errors from Bazel instead of awkward Scala compilation errors when you have Ccala version mismatches).